packages/cojson/src/coValueCore/coValueCore.ts

```diff
@@ -628,6 +739,75 @@ export class CoValueCore {
     }
   }
 
+  #markAsDeleted() {
+    this.isDeleted = true;
+    this.verified?.markAsDeleted();
+  }
+
+  #isDeleteMarkerTransaction(tx: Transaction): boolean {
+    if (tx.privacy !== "trusting") {
+      return false;
+    }
+    if (!tx.meta) {
+      return false;
+    }
+    const meta = safeParseJSON(tx.meta);
+    return meta?.deleted === true;
+  }
+
+  #canAuthorDeleteCoValueAtTime(
+    author: RawAccountID | AgentID,
+    madeAt: number,
+  ):
+    | { ok: true }
+    | {
+        ok: false;
+        reason: DeleteTransactionRejectedError["reason"];
+        message: string;
+      } {
+    if (!this.verified) {
+      return {
+        ok: false,
+        reason: "CannotVerifyPermissions",
+        message: "Cannot verify delete permissions without verified state",
+      };
+    }
+
+    if (this.isGroupOrAccount()) {
+      return {
+        ok: false,
+        reason: "CoValueNotDeletable",
+        message: "Cannot delete Group or Account coValues",
+      };
+    }
+
+    const group = this.safeGetGroup();
+
+    // Today, delete permission is defined in terms of group-admin on the owning group.
+    // If we cannot derive that (non-owned coValues), we reject the delete when verification is required.
```

Comment:
We should never hit this check because we already validated the covalue is not a group or an account, right? I'd add this to the comment